client/lib/format-number-compact/test/index.js

```diff
@@ -1,195 +1,4 @@
-import formatNumberCompact, { formatNumberMetric } from 'calypso/lib/format-number-compact';
-
-describe( 'formatNumberCompact', () => {
```

Comment:
I wonder what would happen if we run these tests against the alternative solution that you're recommending to use 🤔

Reply:
It looks like there are some differences - the Arabic version, for instance, used to be `1٫2 ألف`. It's now `١٫٢ ألف`.

I'm inclined to defer to `Intl.NumberFormat` on this.

Reply:
I'd want to double-check this though, since it might introduce unexpected regressions in some locales.

On top of the @Automattic/i18n team you already pinged, perhaps we could ask a native speaker? IIRC @alshakero could help, but he's on leave. The i18n team is also on a meetup this week AFAIK, so this might take a bit longer than usual.

Reply:
Definitely don't intend to block this work. I just don't feel particularly knowledgeable about this aspect of i18n. The i18n team has historically worked with RTL languages way more, so I hope they will have a better idea.